README.rst

```diff
@@ -2,3 +2,31 @@ colcon-installed-package-information
 ====================================
 
 Extensions for `colcon-core <https://github.com/colcon/colcon-core>`_ to inspect packages which have already been installed.
+
+Overview
+--------
+
+These colcon extensions provide a mechanism which can be used for getting information about packages outside of the workspace, which have already been built and installed prior to the current operation.
+In general, it works similarly to and is based on the `PackageDiscoveryExtensionPoint <https://colcon.readthedocs.io/en/released/developer/extension-point.html#packagediscoveryextensionpoint>`_ and `PackageAugmentationExtensionPoint <https://colcon.readthedocs.io/en/released/developer/extension-point.html#packageaugmentationextensionpoint>`_ extensions provided by ``colcon-core``.
+
+Differences
+-----------
+
+Installed packages don't generally have a single directory which stores the package content and metadata.
+This set of extensions store the "prefix" under which the package resides rather than the package directory, meaning many packages will likely share the same ``path`` attribute value.
+
+Recursively crawling an entire system or even selective subdirectories to look for installed packages could be very slow, so this process also deviates from the Discover -> Identify -> Augment pipeline used in ``colcon-core``.
+Rather than attempting identification on prospective package locations, the discovery phase generally loads a list of installed packages from a database of some kind.
```

Comment:
Would it be good to list an example, or what typically happens (e.g. pulling from the ament prefix path chain/colcon index)? Or is it better to leave this vague?

Reply:
87ddf9a